honeycomb-kernels/src/grisubal/kernel.rs

```diff
@@ -509,11 +508,39 @@ pub(super) fn insert_intersections<T: CoordsFloat>(
         });
 
     // b.
-    for (edge_id, vs) in &mut edge_intersec {
+    // FIXME: minimize allocs & redundant operations
```

Comment:
Would you happen to have an idea of the overestimate?

Reply:
I'm not 100% sure, I added the comment because I didn't think a lot about the impl and just wrote it as it came to me. I hope there is a better way to implement this because the number of iterators & `.collect` calls seems atrocious. 

The good news is that there is no change of performance with this impl in sequential, so it's at least as good as the previous impl while enabling parallelization of the next loop.